scripts/envsetup.sh

```diff
@@ -5,9 +5,9 @@ MACHINE="$1"
 BUILDDIR="build"
 DISTRO="poky-sota-systemd"
 BASE_CONF="local.conf.base.append"
-declare -A supported_distros=( ["poky-sota-systemd"]="local.conf.systemd.append" ["poky-sota"]="local.conf.base.append" )
+declare -A supported_distros=( ["poky-sota-systemd"]="local.conf.systemd.append" ["poky-sota"]="local.conf.base.append" ["poky-sota-nonostree"]="local.conf.nonostree.append")
```

Comment:
From the list poky-sota-systemd, poky-sota and poky-sota-nonostree, one would expect that poky-sota-nonostree does not use systemd.

In the current state, I'm not sure why we actually need a distro and not just add the systemd specific features to local.conf.

If we do want to have such distro, I agree with @mike-sul: maybe we should also use some of the options from sota.conf.inc, especially for reproducible builds. It's always a nice property and it should help sharing cached work with ostree builds.

Reply:
Another reason to use distro is to keep a uniform approach to generate local.conf in the `envsetup.sh`,  introduced by @mike-sul 

For me, it also looks better to add some exceptional case in` envsetup.sh` and remove unnecessary distro(poky-sota-nonostree).

Just for reminder, I still want to use "poky" distro without inheriting `poky-sota`